lib/openpgp_crypt_rsa.php

```diff
@@ -228,43 +230,72 @@ static function try_decrypt_session($key, $edata) {
   }
 
   static function crypt_rsa_key($mod, $exp, $hash='SHA256') {
-    $rsa = new Crypt_RSA();
-    $rsa->setSignatureMode(CRYPT_RSA_SIGNATURE_PKCS1);
-    $rsa->setHash(strtolower($hash));
-    $rsa->modulus = new Math_BigInteger($mod, 256);
-    $rsa->k = strlen($rsa->modulus->toBytes());
-    $rsa->exponent = new Math_BigInteger($exp, 256);
-    $rsa->setPublicKey();
-    return $rsa;
+
+    return Crypt_RSA::loadPublicKey([
+        'e' => new Math_BigInteger($exp, 256),
+        'n' => new Math_BigInteger($mod, 256),
+    ])
+        ->withPadding(CRYPT_RSA_SIGNATURE_PKCS1)
+        ->withHash(strtolower($hash));
+    ;
+
   }
 
   static function convert_key($packet, $private=false) {
     if(!is_object($packet)) $packet = OpenPGP_Message::parse($packet);
     if($packet instanceof OpenPGP_Message) $packet = $packet[0];
 
-    $mod = $packet->key['n'];
+
+
     $exp = $packet->key['e'];
     if($private) $exp = $packet->key['d'];
     if(!$exp) return NULL; // Packet doesn't have needed data
 
-    $rsa = self::crypt_rsa_key($mod, $exp);
+    /**
+     * @see https://github.com/phpseclib/phpseclib/issues/1113
+     * Primes and coefficients now use BigIntegers.
+     **/
 
     if($private) {
         /**
-         * @see https://github.com/phpseclib/phpseclib/issues/1113
-         * Primes and coefficients now use BigIntegers.
-         **/
-        //set the primes
-        if($packet->key['p'] && $packet->key['q'])
-            $rsa->primes = array(
-                1 => new Math_BigInteger($packet->key['p'], 256),
-                2 => new Math_BigInteger($packet->key['q'], 256)
-            );
-        // set the coefficients
-        if($packet->key['u']) $rsa->coefficients = array(2 => new Math_BigInteger($packet->key['u'], 256));
+         * @see https://github.com/phpseclib/phpseclib/issues/1711
+         */
+        $xml = sprintf(
+            <<<XMLRSA
+            <RSAKeyValue>
+                <Modulus>%s</Modulus>
+                <Exponent>%s</Exponent>
+                <P>%s</P>
+                <Q>%s</Q>
+                <!--
+                <DP></DP>
+                <DQ></DQ> -->
```

Comment:
these two  part are not given by the `nepqud` parameters,  causing latter internal error in phpsclib as they have not been set 

my understanding is that they can be calculated  as they are only chinese remainder optimization , should we computer them here once and for all , or am I going in the wrong direction ?